Source/UI/NewUI/PaymentForm/Factory/PaymentFormFactory.swift

```diff
@@ -4,17 +4,18 @@ import Checkout
 public typealias Card = Checkout.Card
 
 public struct PaymentFormFactory {
-  public static func getPaymentFormViewController(environment: Environment,
-                                                  billingFormData: BillingForm?,
-                                                  paymentFormStyle: PaymentFormStyle?,
-                                                  billingFormStyle: BillingFormStyle?,
-                                                  supportedSchemes: [Card.Scheme]) -> UIViewController {
-    let cardValidator = CardValidator(environment: environment.checkoutEnvironment)
+
+  public static func buildViewController(configuration: PaymentFormConfiguration,
+                                         style: PaymentStyle) -> UIViewController {
+    let cardValidator = CardValidator(environment: configuration.environment.checkoutEnvironment)
+    let sessionCorrelationID = UUID().uuidString
+    let logger = FramesEventLogger(environment: configuration.environment, getCorrelationID: { sessionCorrelationID })
```

Comment:
I think this `sessionCorrelationID` might need to be moved into a static private property, just to make sure it persists accurately and doesn't turn into a `nil` 🤔. Will run some local QA to check this.

Reply:
Would be also good to check existing frames 3.5.3 behaviour.

Reply:
Yes, this is the behaviour that was happening before, I have checked how it was done and found it to be excessively convoluted for no apparent reason.

The correlation ID was not persisted anywhere, it was just kept in memory via a complex chain of `logger -> service -> log manager -> static UUID string`. You can review source of it all inside `LogManager.swift`.

Only behaviour change is that I'm not enforcing lowercasing as it was previously done. Would be quite curious to know why we would willingly reduce our uniqueness chance of the correlation id?

Reply:
Additionally, the PaymentForm (via its VC & VM) are the core owners of the payment journey. Anything presented after them will be able to share the state they have (ie: logger correlation id).

If they are dismissed (ie: user cancels intention to make payment) then in this current approach we do generate a new correlation ID for their second start of the journey.

Big ❗ here, the event for payment billing form being shown is marked as unique, so will only be reported once per correlation id. In practice here, that means if we persist that id we lose insights:
- open payment form - 🟢 
- change mind and cancel 🔴 
- open payment form 🔴 
- open billing form 🟢 
- change mind and cancel all 🔴 
- open payment form 🔴 
- open billing form 🔴 
- make payment 🟢 

There is potential for large time gaps between these events as the user may be doing online shopping or generally interacting further with the application. To us it may appear that from start to end it took user 20 minutes, when in reality they dropped twice at different levels of the journey.

Reply:
Good point we can track all the red events as well with a unique correlation id, But we then we need to track cancel event if we want to make more sense with our logs ( otherwise we can just have presented logs fired up more times  for first release @patrick-hoban-cko is that ok with you ? ).